src/constraints/capacity.jl

```diff
@@ -90,26 +143,131 @@ function add_capacity_constraints!(
+    assets_profile_times_capacity_in_with_binary_part1 =
+        model[:assets_profile_times_capacity_in_with_binary_part1] = [
+            if row.asset ∈ Ai && row.asset ∈ Asb
+                @expression(
+                    model,
+                    profile_aggregation(
+                        Statistics.mean,
+                        graph[row.asset].rep_periods_profiles,
+                        (:availability, row.rp),
+                        row.timesteps_block,
+                        1.0,
+                    ) *
+                    (graph[row.asset].initial_capacity + graph[row.asset].investment_limit) *
+                    row.is_charging
+                )
+            else
+                @expression(
+                    model,
+                    profile_aggregation(
+                        Statistics.mean,
+                        graph[row.asset].rep_periods_profiles,
+                        (:availability, row.rp),
+                        row.timesteps_block,
+                        1.0,
+                    ) *
+                    (graph[row.asset].initial_capacity) *
+                    row.is_charging
+                )
+            end for row in eachrow(dataframes[:highest_in])
+        ]
+
+    assets_profile_times_capacity_in_with_binary_part2 =
+        model[:assets_profile_times_capacity_in_with_binary_part2] = [
+            if row.asset ∈ Ai && row.asset ∈ Asb
+                @expression(
+                    model,
+                    profile_aggregation(
+                        Statistics.mean,
+                        graph[row.asset].rep_periods_profiles,
+                        (:availability, row.rp),
+                        row.timesteps_block,
+                        1.0,
+                    ) * (
+                        graph[row.asset].initial_capacity * row.is_charging +
+                        graph[row.asset].capacity * assets_investment[row.asset]
+                    )
+                )
+            end for row in eachrow(dataframes[:highest_in])
+        ]
+
     ## Capacity limit constraints (using the highest resolution)
     # - Maximum output flows limit
     model[:max_output_flows_limit] = [
-        @constraint(
-            model,
-            outgoing_flow_highest_out_resolution[row.index] ≤
-            assets_profile_times_capacity_out[row.index],
-            base_name = "max_output_flows_limit[$(row.asset),$(row.rp),$(row.timesteps_block)]"
-        ) for row in eachrow(dataframes[:highest_out]) if
+        if row.asset ∉ Asb
+            @constraint(
+                model,
+                outgoing_flow_highest_out_resolution[row.index] ≤
+                assets_profile_times_capacity_out[row.index],
+                base_name = "max_output_flows_limit[$(row.asset),$(row.rp),$(row.timesteps_block)]"
+            )
+        end for row in eachrow(dataframes[:highest_out]) if
```

Comment:
I think here you want to change the `if` to the end of the `for`, like the other `if`?